### PR TITLE
Use an R option or env variable to turn on/off Connect content cache

### PR DESCRIPTION
**Pull request by juliasilge**
Addresses #635

### CODE REVIEW

File: R/board_rsconnect.R [hunk 1]
```diff
@@ -549,20 +550,33 @@ rsc_user_name <- function(board, guid) {
...
     list()
   }
 }
-update_cache <- function(path, key, value) {
-  cache <- read_cache(path)
-  cache[[key]] <- value
-  write_yaml(cache, path)
 
+update_cache <- function(path, key, value) {
+  if (use_cache()) {
```
Comment (juliasilge) [inline R/board_rsconnect.R:561]: I am checking the cache option in _both_ `read_cache()` and `update_cache()` even though the second calls the first so we avoid writing to the cache here in a couple of lines ⬇️

**Comment by juliasilge** [comment]
This PR also does some code styling on `board_rsconnect.R`.

File: R/board_rsconnect.R [hunk 2]
```diff
@@ -549,20 +550,33 @@ rsc_user_name <- function(board, guid) {
...
+    cache <- read_cache(path)
+    cache[[key]] <- value
+    write_yaml(cache, path)
+  }
```
Comment (juliasilge) [inline R/board_rsconnect.R:565]: I am checking the cache option in _both_ `read_cache()` and `update_cache()` even though the second calls the first so we avoid writing to the cache here

File: R/board_rsconnect.R [hunk 3]
```diff
@@ -549,20 +550,40 @@ rsc_user_name <- function(board, guid) {
...
   value
 }
 
+clear_cache <- function(board,
+                        cache = c("content-cache.yml", "users-cache.yml")) {
+  path <- fs::path(board$cache, cache)
+  cli::cli_inform("Deleting {.file {path}}")
+  fs::file_delete(path)
+}
```
Comment (juliasilge) [inline R/board_rsconnect.R:574]: I added this new little function to delete the cache files. Right now, I have not exported or documented it, which makes it like a helper function for us to use in development and sort of secondarily, something for us to tell people to use when they get stuck. People would use it like:

```r
pins:::clear_cache(b, "content-cache.yml")
```
Thoughts? Should this be a regular exported function instead?

Comment (juliasilge) [reply R/board_rsconnect.R:574]: If we do really end removing these caches, then this function would need to be deprecated (if it were exported).

Comment (juliasilge) [reply R/board_rsconnect.R]: I decided to export and document it, and we can deal with that later.

File: R/board_rsconnect.R [hunk 4]
```diff
@@ -549,20 +550,77 @@ rsc_user_name <- function(board, guid) {
...
+#' caches. You can manually delete a broken cache via `clear_connect_cache()`.
+#'
+#' @details The `use_connect_cache()` function uses the following method for
+#' detecting whether or not to use the content and user caches for Connect:
+#'
+#' 1. If the `pins_connect_cache` option is set to `FALSE`, `FALSE` is returned.
+#' 2. If the `PINS_CONNECT_CACHE` environment variable is set to `"false"`,
+#' `"FALSE"`, or `"False"`, `FALSE` is returned
+#' 3. In all other cases (including empty or `NULL` values for either of the
+#' above), `TRUE` is returned
```
Comment (juliasilge) [inline R/board_rsconnect.R:592]: The logic in the function is more complicated in order to give some helpful error messages for bad values. Although maybe I should reword this, in that case?

**Review by machow — APPROVED** [review]
LGTM!

**Comment by juliasilge** [comment]
Closed by #667

**Pull request closed without merging**